src/app/icd/client/DefaultICDStorageKeyDelegateImpl.cpp

```diff
@@ -0,0 +1,34 @@
+/*
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#include "DefaultICDStorageKeyDelegateImpl.h"
+
+namespace chip {
+namespace app {
+
+void DefaultICDStorageKeyDelegateImpl::Init(size_t aMaxKeyCounter)
+{
+    mMaxKeyCounter = aMaxKeyCounter;
+}
+
+StorageKeyName DefaultICDStorageKeyDelegateImpl::GetKey(size_t index)
+{
+    return StorageKeyName::Formatted("%x", static_cast<unsigned>(index));
```

Comment:
This is not scoped enough. This will just cause "0..1........f....10" and is not scoped to per-fabric.

Reply:
removed, thanks